javascript/code-challenges/challenges-07.test.js

```diff
@@ -0,0 +1,359 @@
+
+Write a function named howMuchPencil that takes in a string, as written on the side of a pencil.
+
+As you sharpen the pencil, the string will become shorter and shorter, starting by removing the first letter.
+
+Your function should use slice within a loop and return an array of each successive string result from losing letters to the sharpener, until nothing is left.
+
+For example, if the input is 'Welcome', the output will be:
+['Welcome', 'elcome', 'lcome', 'come', 'ome', 'me', 'e', ''].
+------------------------------------------------------------------------------------------------ */
+
+const howMuchPencil = (str) => {
+  let result = [];
+  for (let i = str.length; i <= 0; i++) {
+    result += str[i];
+    result.slice(i);
+  }
+  return result;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 5
+
+Write a function name wordsToCharList that, given a string as input, returns a new array where every element is a character of the input string.
+
+For example, wordsToCharList('gregor') returns ['g','r','e','g','o','r'].
+------------------------------------------------------------------------------------------------ */
+
+const wordsToCharList = (arr) => {
+  return arr.split('');
+};
+
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 6
+
+You are making a grocery list for ingredients needed in the gruffalo crumble recipe, below. Rather than taking the entire recipe, you only want a list of the item names.
+
+Write a function named listFoods that takes in the recipe and returns a new array of the food items without any amount or units. Just the name. For example, '1 cup flour' will return 'flour'.
+
+Use slice for this function, maybe more than once. The Array.indexOf() method may also be helpful.
+
+Do not use split for this function.
+------------------------------------------------------------------------------------------------ */
+
+const gruffaloCrumble = {
+  name: 'How to make a Gruffalo Crumble',
+  ingredients: [
+    '1 medium-sized Gruffalo',
+    '8 pounds oats',
+    '2 pounds brown sugar',
+    '4 pounds flour',
+    '2 gallons pure maple syrup',
+    '16 cups chopped nuts',
+    '1 pound baking soda',
+    '1 pound baking powder',
+    '1 pound cinnamon',
+    '6 gallons melted butter',
+    '2 gallons fresh water',
+  ],
+  steps: [
+    'Pre-heat a large oven to 375',
+    'De-prickle the gruffalo',
+    'Sprinkle with cinnamon, sugar, flour, and nuts',
+    'Mix until evenly distributed',
+    'Grease a 3-foot x 3-foot casserole dish',
+    'Combine gruffalo compote with water to maintain moisture in the oven',
+    'Fold together remaining ingredients to make the crisp',
+    'Spread the crisp evenly over the gruffalo mixture',
+    'Bake for 12-15 hours',
+  ]
+};
+
+
+const listFoods = (recipe) => {
+  let result = [];
+  recipe.ingredients.indexOf(-1).slice(-1);
+  return result;
```

Comment:
So right now it looks like you're trying to cut off the last word at the end of the string. A couple problems there
1.  Items are two words and some are one. It's inconsistent. You want "brown sugar" not just sugar. But if you remove the last two words of every sentence, then you'd also end up with "pound oats" which is not what you want.
What is consistent though, its the amount of everything is always two words. It's always a number , and then the metric (8 pounds, 16 cups). So if you removed the first two words of the string, then you'd be left exclusively with the ingredients you want to return.
2. Recipe is an object. You need to access the array "ingredients" from inside the object "recipe".
3. Currently ingredients is an array of strings, but in order to manipulate the data you probably need to turn those strings into an array. Good news is you already know how to do that cause you did it in challenge 5 just up above.